perfkitbenchmarker/benchmarks/object_storage_service_benchmark.py

```diff
@@ -169,19 +169,42 @@ def _JsonStringToPercentileResults(results, json_input, metric_name,
         metadata))
 
 
-# TODO: Add api-based benchmarks for Azure here and then consider splitting this
-# into multiple functions, one for each scenario.
-def S3orGCSApiBasedBenchmarks(results, metadata, vm, storage, test_script_path,
-                              bucket_name, regional_bucket_name=None):
-    """This runs a collection of api-based benchmarks for s3 or GCS.
+def _MakeAzureCommandSuffix(account_name, account_key, for_cli):
+  """ This function returns a suffix for Azure command.
+
+  Args:
+    account_name: The name of the Azure storage account.
+    account_key: The key to access the account.
+    for_cli: If true, the suffix can be passed to the Azure cli tool; if false,
+      the suffix created will be used to call our own test script for api-based
+      tests.
+
+  returns:
+    A string represents a command suffix.
+  """
+
+  if for_cli:
+    return (' -a %s -k %s') % (account_name, account_key)
+  else:
+    return (' --azure_account=%s --azure_key=%s') % (account_name, account_key)
+
+
+def ApiBasedBenchmarks(results, metadata, vm, storage, test_script_path,
+                       bucket_name, regional_bucket_name=None,
+                       azure_command_suffix=None):
+    """This function contains all api-based benchmarks. It uses the value of
```

Comment:
The first line is a short description of the method. More detailed go to next section. eg.
This function contains all api-based benchmarks.
<empty line>
It uses the value of ...